### PR TITLE
Add support for screener-wide banners (CO SNAP warning)

### PR DESCRIPTION
**Pull request by jmejia**
## Context & Motivation

- Implements frontend for MFB-398: Add configurable system-wide banner message system
- Related PR: https://github.com/MyFriendBen/benefits-api/pull/1203
- Enables display of urgent notifications (e.g., SNAP benefits hold) across all pages
- Supports multiple banners with priority ordering and per-white-label configuration
- Banners are collapsible to reduce visual clutter while keeping important info accessible
- Supports very basic markdown in banners but is setup to easily expand in the future (ie react-markdown)

banner_demo

### New Components & Utilities

- **SystemBanner component** (`src/Components/Common/SystemBanner/SystemBanner.tsx`)
  - Displays configurable banner messages on all pages
  - Collapsed by default with inline More/Less toggle links
  - Supports multiple banners sorted by priority
  - Filters to only show enabled banners
  - Independent expand/collapse state per banner

- **parseMarkdown utility** (`src/utils/parseMarkdown.tsx`)
  - Reusable markdown parser for banner content
  - Supports `**bold**` syntax
  - Auto-detects URLs and converts to clickable links with `target="_blank"`
  - Preserves line breaks in multi-line content

### Tests

- **SystemBanner.test.tsx**: 9 comprehensive tests covering:
  - Empty/disabled banner handling
  - Collapse/expand functionality
  - Multiple banner rendering and priority ordering
  - Markdown parsing (bold text, clickable links)
  - Independent state management per banner

- **parseMarkdown.test.tsx**: 14 unit tests covering:
  - Plain text, bold syntax, URLs
  - Multiple URLs and bold segments
  - Line break preservation
  - Edge cases (empty content, URLs at start/end, etc.)

## Testing

- **Configuration updates needed**: None (uses existing config API endpoint)
- **Environment variables/settings to add**: None
- **Manual testing steps**:
  1. Start frontend dev server
  2. Navigate to Colorado white label
  3. Verify banner appears at top of all pages (collapsed by default)
  4. Click "More" to expand - should show full SNAP message with bold text and clickable link
  5. Click "Close" to collapse again
  6. Test on mobile width - banner should stay within same width as progress bar

- **Automated tests**:
  ```bash
  npm test -- src/Components/Common/SystemBanner/SystemBanner.test.tsx --watchAll=false
  npm test -- src/utils/parseMarkdown.test.tsx --watchAll=false
  ```

## Deployment

- No special deployment steps needed
- Component will automatically display banners configured in backend white label configs
- To enable/disable banners: Update `banner_messages` in white label config and run `add_config` management command

## Notes for Reviewers

- **Design decisions**:
  - Generic component name (SystemBanner not SNAPBanner) for reusability beyond SNAP notifications
  - Markdown syntax in config (`**bold**`) instead of component-specific logic for flexibility
  - Extracted parseMarkdown to utility for reusability across codebase
  - Collapsed by default to minimize visual impact while keeping info accessible

- **Future considerations**:
  - Could extend markdown parser to support additional syntax (lists, headers, etc.) if needed
  - Banner dismiss/persist state could be added with localStorage if desired

- **Related backend PR**: [Link to backend PR with banner_messages config changes]

## Summary by CodeRabbit

* **New Features**
  * Added system banners: per-banner enable/disable, priority ordering, collapsed default with localized More/Less toggle, template variable rendering, and themed content.

* **Utilities**
  * New markdown parser for banner content supporting bold text, clickable themed links, and preserved line breaks.

* **Tests**
  * Comprehensive tests covering rendering, interactions, formatting, link attributes, ordering, and context behavior.

### CODE REVIEW

**Comment by jmejia** [comment]
@coderabbitai review

File: src/utils/parseMarkdown.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * Supported features:
+ * - **text** for bold
+ * - Auto-detection of URLs (https://...) and converts them to clickable links
+ * - Preserves line breaks
+ *
+ * @param content - The text content to parse
+ * @param primaryColor - Color for links
+ * @returns React nodes with formatted content
+ */
+export const parseMarkdown = (content: string, primaryColor: string): React.ReactNode => {
```
Comment (jmejia) [inline src/utils/parseMarkdown.tsx]: At some point we should consider using a real markdown parsing library but this felt like a lighter weight solution. It should be setup in a way that replacing should be easy in the future.
